torch/csrc/jit/script/compiler.cpp

```diff
@@ -845,15 +747,16 @@ struct to_ir {
                   // never create a Method for the closure
       popFrame(/*ends_def=*/true);
     }
+    AT_ASSERT(block->outputs().size() == 1);
+    // Set the type of the closure equal to its return type.
+    // Since the closures are not-first class yet in the type system,
+    // we use their output type to correctly type forks of closures,
+    // and after the graph is transformed to SSA ensure that forked closures
+    // are actually being invoked on a closure and not an arbitrary value.
+    closure_node->output()->setType(block->outputs().at(0)->type());
     std::shared_ptr<Graph> subgraph;
-    Value* context;
-    std::tie(subgraph, context) = lambdaLift(block);
-    runCleanupPasses(subgraph);
-    closure_node->eraseBlock(0);
-    closure_node->g_(attr::Subgraph, std::move(subgraph));
-    auto tup =
-        graph->insertNode(graph->createTuple({closure_node->output(), context}))
-            ->output();
+    auto tup = graph->insertNode(graph->createTuple({closure_node->output()}))
```

Comment:
Why is a tuple node being created _now_ and not when we are lowering the closure? We should have a prim::Closure node to represent this temporarily.